src/zenml/integrations/kserve/steps/kserve_deployer.py

```diff
@@ -0,0 +1,192 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the KServe Deployer step."""
+import os
+from typing import cast
+
+from zenml.artifacts.model_artifact import ModelArtifact
+from zenml.environment import Environment
+from zenml.integrations.kserve.model_deployers.kserve_model_deployer import (
+    DEFAULT_KSERVE_DEPLOYMENT_START_STOP_TIMEOUT,
+    KServeModelDeployer,
+)
+from zenml.integrations.kserve.services.kserve_deployment import (
+    KServeDeploymentConfig,
+    KServeDeploymentService,
+)
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.steps import (
+    STEP_ENVIRONMENT_NAME,
+    BaseStepConfig,
+    StepEnvironment,
+    step,
+)
+from zenml.steps.step_context import StepContext
+from zenml.utils import io_utils
+
+logger = get_logger(__name__)
+
+
+class KServeDeployerStepConfig(BaseStepConfig):
+    """KServe model deployer step configuration.
+
+    Attributes:
+        service_config: KServe deployment service configuration.
+        secrets: a list of ZenML secrets containing additional configuration
+            parameters for the KServe deployment (e.g. credentials to
+            access the Artifact Store where the models are stored). If supplied,
+            the information fetched from these secrets is passed to the KServe
+            deployment server as a list of environment variables.
+    """
+
+    service_config: KServeDeploymentConfig
+    timeout: int = DEFAULT_KSERVE_DEPLOYMENT_START_STOP_TIMEOUT
+
+
+@step(enable_cache=False)
+def kserve_model_deployer_step(
+    deploy_decision: bool,
+    config: KServeDeployerStepConfig,
+    context: StepContext,
+    model: ModelArtifact,
+) -> KServeDeploymentService:
+    """KServe model deployer pipeline step.
+
+    This step can be used in a pipeline to implement continuous
+    deployment for an ML model with KServe.
+
+    Args:
+        deploy_decision: whether to deploy the model or not
+        config: configuration for the deployer step
+        model: the model artifact to deploy
```

Comment:
IMO it's cleaner to have all artifacts together at the start of the argument list

```suggestion
    model: ModelArtifact,
    config: KServeDeployerStepConfig,
    context: StepContext,
) -> KServeDeploymentService:
    """KServe model deployer pipeline step.

    This step can be used in a pipeline to implement continuous
    deployment for an ML model with KServe.

    Args:
        deploy_decision: whether to deploy the model or not
        model: the model artifact to deploy
        config: configuration for the deployer step
```